lib/formtastic/inputs/base/validations.rb

```diff
@@ -138,7 +138,8 @@ def required?
           if validations?
             validations.select { |validator|
               if validator.options.key?(:on)
-                return false if (validator.options[:on] != :save) && ((object.new_record? && validator.options[:on] != :create) || (!object.new_record? && validator.options[:on] != :update))
+                validator_on = validator.options[:on].is_a?(Array) ? validator.options[:on] : [validator.options[:on]]
```

Comment:
```ruby
validator.options[:on].is_a?(Array) ? validator.options[:on] : [validator.options[:on]]
```
is the same as
```ruby
Array(validator.options[:on])
```
`Array` is also a function. See http://docs.ruby-lang.org/en/2.2.0/Kernel.html#method-i-Array.